src/cargo/core/resolver/mod.rs

```diff
@@ -687,7 +721,7 @@ impl Context {
                                              feature)));
                 }
             }
-            ret.push((dep, base));
+            ret.push((Rc::new(dep.clone()), base));
```

Comment:
'k, this is done.

Another way to avoid this would be if there were an `Rc` equivalent to C++11's [`template<class Y> shared_ptr(const shared_ptr<Y>& r, T *ptr)`](http://en.cppreference.com/w/cpp/memory/shared_ptr/shared_ptr) constructor that holds a reference to an owning object but dereferences to something it owns.